pallets/pooled-staking/src/calls.rs

```diff
@@ -0,0 +1,459 @@
+            candidate,
+            delegator,
+            from: pool,
+            pending: leaving_stake.0,
+            released: dust,
+        });
+
+        Ok(().into())
+    }
+
+    pub fn execute_pending_operations(
+        origin: OriginFor<T>,
+        operations: Vec<PendingOperationQuery<T::AccountId, T::BlockNumber>>,
+    ) -> DispatchResultWithPostInfo {
+        // We don't care about the sender.
+        let _ = ensure_signed(origin)?;
+
+        for (index, query) in operations.into_iter().enumerate() {
+            // We deconstruct the query and find the balance associated with it.
+            // If it is zero it may not exist or have been executed before, thus
+            // we simply skip it instead of erroring.
+            let PendingOperationQuery {
+                delegator,
+                operation,
+            } = query;
+
+            let value = PendingOperations::<T>::get(&delegator, &operation);
+
+            if Zero::is_zero(&value) {
+                continue;
+            }
+
+            match &operation {
+                PendingOperationKey::JoiningAutoCompounding {
+                    candidate,
+                    at_block,
+                } => {
+                    ensure!(
+                        T::JoiningRequestFilter::can_be_executed(&candidate, &delegator, *at_block),
+                        Error::<T>::RequestCannotBeExecuted(index as u16)
+                    );
+
+                    Self::execute_joining(
+                        candidate.clone(),
+                        delegator.clone(),
+                        TargetPool::AutoCompounding,
+                        Shares(value),
+                    )?;
+                }
+                PendingOperationKey::JoiningManualRewards {
+                    candidate,
+                    at_block,
+                } => {
+                    ensure!(
+                        T::JoiningRequestFilter::can_be_executed(&candidate, &delegator, *at_block),
+                        Error::<T>::RequestCannotBeExecuted(index as u16)
+                    );
+
+                    Self::execute_joining(
+                        candidate.clone(),
+                        delegator.clone(),
+                        TargetPool::ManualRewards,
+                        Shares(value),
+                    )?;
+                }
+                PendingOperationKey::Leaving {
+                    candidate,
+                    at_block,
+                } => {
+                    ensure!(
+                        T::LeavingRequestFilter::can_be_executed(&candidate, &delegator, *at_block),
+                        Error::<T>::RequestCannotBeExecuted(index as u16)
+                    );
+
+                    Self::execute_leaving(candidate.clone(), delegator.clone(), Shares(value))?;
+                }
+            }
+
+            PendingOperations::<T>::set(&delegator, &operation, Zero::zero());
```

Comment:
Could we simply kill the entry?

Reply:
We could, is it better than setting to zero?

Reply:
I guess for storage purposes it is.